### PR TITLE
Feat: Non chargeable licence flow - Refactored

### PR DESCRIPTION
**Pull request by Stephan-Hough**
Feat/water 2698 non charge flow

Updates the UI to enable the non chargeable licence flow from the charge
information tab.

Makes the start-date form dual purpose, now capturing either the start
date of a new charge version, or the effective date for a non chargeable
charge version.

Depends on DEFRA/water-abstraction-service#1144

### CODE REVIEW

File: test/internal/modules/charge-information/controllers/non-chargeable.js [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    ]
+  },
+  yar: {
+    get: sandbox.stub()
+  },
+  setDraftChargeInformation: sandbox.stub(),
+  clearDraftChargeInformation: sandbox.stub()
+});
+
+// const getReadableDate = str => moment(str).format('D MMMM YYYY');
```
Comment (DGajewska) [inline test/internal/modules/charge-information/controllers/non-chargeable.js:74]: You forgot to remove this comment

File: src/internal/modules/charge-information/routes/non-chargeable.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        }
+      },
+      validate: {
+        params: {
+          licenceId: VALID_GUID
+        },
+        query: {
+          form: VALID_GUID.optional(),
+          start: Joi.boolean().optional(),
+          returnToCheckData: Joi.number().default(0).allow(0, 1)
```
Comment (DGajewska) [inline src/internal/modules/charge-information/routes/non-chargeable.js:30]: I feel like `start` and `returnToCheckData` should be consisten(ie a boolean or an integer) as they are serving the same prupose. I used an integer for `returnToCheckData` to match the pattern in the billing module `back` pattern. I'm happy to chat this over, if you'd like :)

File: src/internal/modules/charge-information/lib/routing.js [hunk 3]
```diff
@@ -1,10 +1,27 @@
...
+
+exports.getCheckData = createUrl('check');
+exports.getCreateBillingAccount = createUrl('billing-account/create');
+exports.getNonChargeableReason = createUrl('non-chargeable-reason');
+exports.getReason = createUrl('create');
+exports.getStartDate = createUrl('start-date');
+exports.getSelectBillingAccount = createUrl('billing-account');
+exports.getUseAbstractionData = createUrl('use-abstraction-data');
+exports.getEffectiveDate = createUrl('effective-date');
+exports.getNonChargeableReason = createUrl('non-chargeable-reason');
```
Comment (DGajewska) [inline src/internal/modules/charge-information/lib/routing.js:26]: This one is doubled up, first one is on line 20

**Review by DGajewska — CHANGES_REQUESTED** [review]

File: src/internal/modules/charge-information/controllers/charge-information.js [hunk 4]
```diff
@@ -123,30 +119,37 @@ const postUseAbstractionData = createPostHandler(
...
+    ? routing.getUseAbstractionData(licenceId)
+    : routing.getEffectiveDate(licenceId);
+
+  const invoiceAccountAddress = isChargeable ? draftChargeInformation.invoiceAccount.invoiceAccountAddresses
+    .find(address => address.id === draftChargeInformation.invoiceAccount.invoiceAccountAddress) : null;
 
   const view = {
     ...getDefaultView(request, routing.getUseAbstractionData),
     pageTitle: 'Check charge information',
+    back,
```
[the remark this answers is not in the document]
Comment (Stephan-Hough) [reply src/internal/modules/charge-information/controllers/charge-information.js:135]: Yep that was an obvious mistake oops :-D

**Review by DGajewska — APPROVED** [review]

**Pull request merged**